submissions/Ant-C-tech/oop-exercise/index.js

```diff
@@ -0,0 +1,126 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Ant-C-tech/a-tiny-JS-world
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+    constructor(name, friends) {
+        this.name = name;
+        this.friends = friends;
+        this.templete = `My name is ${this.name}`;
+    };
+
+    getVoice() {
+        let pluralEnding;
+        (this.friends && this.friends.length > 1) ? pluralEnding = 's': pluralEnding = '';
+        return (this.friends) ? [
+            this.templete,
+            `I have friend${pluralEnding}: ${this.friends.join(', ')}`
+        ].join('. ') : [
+            this.templete,
+            'I am looking for friends'
+        ].join('. ');
+    };
+};
+
+class Animal extends Inhabitant {
+    constructor(gender, name, friends, legs) {
+        super(name, friends);
+        this.gender = gender;
+        this.legs = legs;
+    };
+
+    getVoice() {
+        return [
+            `I have ${this.legs} legs`,
+            `My gender is ${this.gender}`,
+            super.getVoice(),
+        ].join('. ');
+    };
+}
+
+class Dog extends Animal {
+    constructor(gender, name, friends, legs) {
+        super(gender, name, friends, legs);
+        this.species = 'dog';
+        this.saying = 'Woof-Woof!';
+        this.legs = 4;
+    };
+
+    getVoice() {
+        return [`I am a ${this.species}`,
+            super.getVoice(),
+            `I want to say you: "${this.saying}".`
+        ].join('. ');
+    };
+}
+
+class Cat extends Animal {
+    constructor(gender, name, friends, legs) {
+        super(gender, name, friends, legs);
+        this.species = 'cat';
```

Comment:
All inhabitants have prop `species` just like they all have e.g. `name`.
Why handling `species` is different from the way you handle `name`?